tests/test_query_extensions.py

```
@@ -204,6 +204,84 @@ def cursor(self):
+    assert result["results"] == [], f"Expected empty semantic result set from fake cursor, got {result}"
+    assert len(fake_cursor.calls) == 1, f"Expected exactly one semantic query execution, got {fake_cursor.calls}"
+    sql, params = fake_cursor.calls[0]
+    assert "e.timestamp >= ?" in sql, f"Expected date filter in semantic SQL, got {sql}"
+    assert params == [b"fake-embedding", "2030-01-01", 10], (
+        f"Expected semantic params to include date filter placeholder, got {params}"
+    )
+
+
+def test_semantic_query_with_is_outbound_filter_applies_shared_where_clause():
+    class FakeCursor:
+        def __init__(self):
```

Ruff (0.15.15)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)